src/installer.ts

```diff
@@ -338,10 +365,57 @@ async function resolveVersionFromManifest(
   }
 }
 
+function evaluateNightlyVersions(
```

Comment:
Looks like the same function is being added in https://github.com/actions/setup-node/pull/605/. I suggested some small changes there to simplify it.